pkg/input/arm/eval.go

```diff
@@ -23,25 +23,28 @@ import (
 // pass data to certain function impls other than what is already in their args.
 type EvaluationContext struct {
 	discoveredResourceSet map[string]struct{}
+	variables             map[string]interface{}
 	funcs                 map[string]armFnImpl
 }
 
-func NewEvaluationContext(discoveredResourceSet map[string]struct{}) *EvaluationContext {
+func NewEvaluationContext(discoveredResourceSet map[string]struct{}, variables map[string]interface{}) *EvaluationContext {
 	evalCtx := &EvaluationContext{
 		discoveredResourceSet: discoveredResourceSet,
+		variables:             variables,
 		funcs: map[string]armFnImpl{
 			"concat":        concatImpl,
 			"resourceGroup": resourceGroupImpl,
 		},
 	}
 	evalCtx.funcs["resourceId"] = evalCtx.resourceIDImpl
+	evalCtx.funcs["variables"] = evalCtx.variablesImpl
```

Comment:
The self-assignment here feels a bit weird, I will look at it in a separate refactor

Reply:
I suppose it is a little strange! It's really either that or a closure AFAIK. Same thing in the end. Of course feel free to refactor but I would also be curious what the aim is? What would we be making more understandable / less of a footgun / something else?